server/service/src/item/item_variant/upsert.rs

```diff
@@ -124,5 +128,38 @@ fn validate(
         return Err(UpsertItemVariantError::ItemDoesNotExist);
     }
 
+    let old_item_variant = ItemVariantRowRepository::new(connection).find_one_by_id(&input.id)?;
+
+    if let Some(old_item_variant) = old_item_variant {
+        if old_item_variant.item_link_id != input.item_id {
```

Comment:
I think for item merge to work correctly, we need to query item by item_link here, and then compare the item ids?

Reply:
[7aa62da](https://github.com/msupply-foundation/open-msupply/pull/5288/commits/7aa62da25be439d59d3afdf2a62eaaae2dad2139)

Done here, but since we're not properly using item_link yet, it's a bit hard to test. We also don't seem to have any item_merge mock items?

I think we can revisit the tests in #5241 ?